File: RentalAttireBackend/RentalAttireBackend/Application/Common/Result.cs
namespace RentalAttireBackend.Application.Common
{
    public class Result<T>
    {
        public T? Data { get; set; }
        public bool IsSuccess { get; set; }
        public string SuccessMessage { get; set; } = string.Empty;
        public string ErrorMessage { get; set; } = string.Empty;
        public static Result<T> Success(T data) => new() { Data = data, IsSuccess = true };
        public static Result<T> Failure(string message) => new() { IsSuccess = false, ErrorMessage = message };
        public static Result<T> SuccessWithMessage(string message) => new() { IsSuccess = true, SuccessMessage = message };
    }
}

File: RentalAttireFrontend/src/app/data/models/result.ts
export class Result<T> {
    private constructor(
        public readonly isSuccess: boolean,
        public readonly data ?: T,
        public readonly errorMessage ?: string,
        public readonly successMessage ?: string
    ) {}

    public static success<T>(data: T) : Result<T> {
        return new Result<T>(true, data, undefined, undefined);
    }

    public static successWithMessage<T>(message : string) : Result<T> {
        return new Result<T>(true, undefined, undefined, message);
    }

    public static failure<T>(message: string) : Result<T> {
        return new Result<T>(false, undefined, message, undefined);
    }
}